### PR TITLE
Replace `base::GUID` with `base::Uuid`

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
Upstream has renamed `base::GUID` to `base::Uuid` to avoid name collisions with some constants in Windows headers. Therefore, all instances of the `base/guid.h` header, as well as anything using the former name, must be changed to use `base::Uuid`, in preparation of the deletion of the legacy name from upstream.

Chromium change:
https://chromium.googlesource.com/chromium/src/+/f800d299bb9374ca8d401c1d326339833cb4280e

commit f800d299bb9374ca8d401c1d326339833cb4280e
Author: Claudio DeSouza <cdesouza@igalia.com>
Date:   Thu Jun 8 03:11:20 2023 +0000

    Delete base/guid.h

    This CL deletes `base/guid.h` from the codebase, as all uses of
    `base::GUID` have been replaced with `base::Uuid`. Additionally, all
    deprecated functions are also getting deleted, as all uses have been
    corrected.

    Bug: 1428566, 1195446

Resolves https://github.com/brave/brave-browser/issues/30990

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by tmancey — APPROVED** [review]
LGTM++

**Review by goodov — APPROVED** [review]

**Review by supermassive — APPROVED** [review]
Wallet core lgtm

File: components/brave_rewards/core/bitflyer/bitflyer.cc [hunk 1]
```diff
@@ -5,9 +5,9 @@
 
 #include <utility>
 
-#include "base/guid.h"
 #include "base/strings/string_number_conversions.h"
 #include "base/strings/stringprintf.h"
+#include "base/uuid.h"
```
Comment (zenparsing) [inline components/brave_rewards/core/bitflyer/bitflyer.cc:10]: Looks unused?

File: components/brave_rewards/core/gemini/gemini.cc [hunk 2]
```diff
@@ -5,9 +5,9 @@
 
 #include <utility>
 
-#include "base/guid.h"
 #include "base/strings/string_number_conversions.h"
 #include "base/strings/stringprintf.h"
+#include "base/uuid.h"
```
Comment (zenparsing) [inline components/brave_rewards/core/gemini/gemini.cc:10]: Unused?

File: components/brave_rewards/core/uphold/uphold.cc [hunk 3]
```diff
@@ -7,10 +7,10 @@
 
 #include <utility>
 
-#include "base/guid.h"
 #include "base/json/json_reader.h"
 #include "base/strings/string_number_conversions.h"
 #include "base/strings/stringprintf.h"
+#include "base/uuid.h"
```
Comment (zenparsing) [inline components/brave_rewards/core/uphold/uphold.cc:13]: Unused?

**Review by zenparsing — APPROVED** [review]
Rewards 👍

File: components/brave_rewards/core/gemini/gemini.cc (hunk 2 above)
Comment (cdesouza-chromium) [reply components/brave_rewards/core/gemini/gemini.cc:10]: Fixed

File: components/brave_rewards/core/uphold/uphold.cc (hunk 3 above)
Comment (cdesouza-chromium) [reply components/brave_rewards/core/uphold/uphold.cc:13]: Fixed. Thanks for catching that.

**Review by kylehickinson — APPROVED** [review]
iOS++

**Pull request merged**